src/graphviz.rs

```diff
@@ -1,9 +1,67 @@
 /// GraphViz visualisation for our SyntaxTree representation
 /// Named syntax nodes are elliptical; anonymous are rectangular
-use std::{fmt, io};
+use std::{borrow::Cow, fmt, io};
 
 use crate::{tree_sitter::SyntaxNode, FormatterResult};
 
+/// We double-escape whitespace (\n and \t) so it is
+/// rendered as the escaped value in the GraphViz output
+fn escape(input: &str) -> Cow<str> {
+    let mut buffer: Option<String> = None;
+
+    let mut start: usize = 0;
+    let length = input.len();
+
+    let append = |buffer: &mut Option<String>, from: &mut usize, to: usize, suffix: &str| {
+        // Allocate buffer only when necessary
+        if buffer.is_none() {
+            // Best case:  length + 1  (i.e., single escaped character in input)
+            // Worst case: length * 3  (i.e., every character needs double-escaping)
+            *buffer = Some(String::with_capacity(length * 2));
```

Comment:
So why multiply with two here (and not by three)?

Reply:
For simplicity, I would just do:

```
        if buffer.is_none() {
            *buffer = Some(String::new());
        }
```

Reply:
Here I'm trying to reduce the number of allocations required as the escaped string gets built up. I used `Option<String>` originally, because I didn't realise that `String::new` _didn't_ allocate. I've just confirmed that now from the manual. As such, I'll get rid of the `Option`, per your other comment, and switch this to `buffer.reserve(whatever)`.

The buffer size is set to `length * 2` as a "happy medium". Given that the strings it encounters will be very short, `length * 3` makes sense too :+1: